aws/src/main/java/org/apache/iceberg/aws/AwsClientFactories.java

```diff
@@ -50,6 +50,14 @@ public static AwsClientFactory defaultFactory() {
   public static AwsClientFactory from(Map<String, String> properties) {
     String factoryImpl = PropertyUtil.propertyAsString(
         properties, AwsProperties.CLIENT_FACTORY, DefaultAwsClientFactory.class.getName());
+    if (PropertyUtil.propertyAsBoolean(properties,
+        AwsProperties.GLUE_LAKEFORMATION_ENABLED,
+        AwsProperties.GLUE_LAKEFORMATION_ENABLED_DEFAULT)) {
+      if (!factoryImpl.equals(LakeFormationAwsClientFactory.class.getName())) {
+        throw new RuntimeException(String.format("LakeFormation is enabled but factoryImpl=%s is provided, " +
```

Comment:
I am thinking what is the best way to go with this. We know not all client factories could work with LF, but what we provide is a default implementation that people can override.

If a vendor would like to add custom credential configurations, they should do `CustomLakeFormationAwsClientFactory extends LakeFormationAwsClientFactory`, so the best way for us to do this check is to first initialize the factory, and check that `factory instanceof LakeFormationAwsClientFactory`, instead of directly checking the impl string.